src/index/hgraph_index.h

```diff
@@ -83,33 +83,32 @@ class HGraphIndex : public Index {
         SAFE_CALL(return this->knn_search(query, k, parameters, filter);)
     }
 
-    // TODO(LHT): implement
     tl::expected<DatasetPtr, Error>
     RangeSearch(const DatasetPtr& query,
                 float radius,
                 const std::string& parameters,
                 int64_t limited_size = -1) const override {
-        return Dataset::Make();
+        SAFE_CALL(return this->range_search(query, radius, parameters, nullptr, limited_size);)
```

Comment:
SAFE_CALL(return this->range_search(query, radius, parameters, nullptr, limited_size));

Reply:
done